kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java

```diff
@@ -381,9 +387,14 @@ private void setWifiInfraStateProperties(GwtWifiNetInterfaceConfig gwtWifiNetInt
         }
 
         ChannelsBuilder channelsBuilder = new ChannelsBuilder();
-        List<Integer> activeChannel = gwtConfig.getChannels();
-        if (activeChannel != null && activeChannel.size() == 1) {
-            channelsBuilder.setActiveChannel(activeChannel.get(0));
+        Optional<WifiAccessPoint> activeAP = wifiInterfaceInfo.getActiveWifiAccessPoint();
```

Comment:
Can this be put in common with the method above `setWifiMasterStateProperties `?